### ISSUE TITLE
Relaunch all failed jobs at once, for a given step 

### ISSUE THREAD

**Issue by vladvisan** [galaxyproject/galaxy#18442]
![image](https://github.com/galaxyproject/galaxy/assets/11979062/935bd71a-cb20-4f43-9aeb-d02341dad2db)
Screenshot taken from @ahmedhamidawan's GCC presentation since my instance doesn’t have this feature yet.

**Related to**

- https://github.com/galaxyproject/galaxy/pull/17413
- I searched and did not find a similar issue

**Description**
- If multiple jobs fail for a same step (collection), it would be nice to be able to relaunch all the failed jobs for that step, at the same time / with one button.
- If not, if there are for example 50 failed jobs, it can be very tedious.
- I’m not sure how often such big (>=50) collections are, but it could happen, potentially of even larger size (potentially hundreds or thousands of datasets).

**Adjacent ideas**
- Handle the cases where there is a mix of successful jobs, failed jobs, running jobs, waiting-to-be-run jobs, ..
- One talk at GCC mentioned a « multi-select datasets » option when launching a tool, maybe the logic or page could be re-used/pre-populated ?
- Maybe allow a multiple-choice checkbox of which jobs to re-execute, by default all selected, with a button to turn them off
- Maybe even a regex to select the jobs to be re-executed, maybe re-using the [collection filter operation](https://usegalaxy.org/?tool_id=__FILTER_FROM_FILE__&version=latest)
- Should also (as usual) include the « Resume dependencies from this job ? » Additional Option
- In the screenshot's workflow, one could just relaunch the previous step as it would in turn relaunch all the failed jobs of the last step. But this solution wouldn't work for a typical workflow (and it is inefficient even when it does work).
- Let the user change the step's tool's version before executing - but keep the rest of the invocation ? 
    - Might be necessary to resolve the underlying error. Not always, sometimes an external resource was unavailable and relaunching the exact same tool/dataset combo works the second time around
    -  Need to be careful about reproducibility. Might need to duplicate/fork a history, or create a new invocation and combine with https://github.com/galaxyproject/galaxy/pull/4690 ?

**Labels**
- feature-request, area/UI-UX, and maybe area/workflows and area/backend

**Comment by vladvisan** [comment]
Thanks for the feedback.

**_I would say the most common thing to do is to re-run a single job, this is the default behavior now, and I think that should remain that way._**
Good point.

**_you can select the input collection today and all jobs will re-run_**
- I must have missed something, I tried to do this, but I was not able to see a rerun/"recycle" button for the collection, only for the individual datasets?
- ![image](https://github.com/galaxyproject/galaxy/assets/11979062/198c8cc8-3653-4339-8cc0-29edd0a01e59)
- I also tried to manually modify the rerun URL of a dataset https://usegalaxy.org/tool_runner/rerun?id=xxxx , and replace the id with the collection's id, but I got a "You are not allowed to access this dataset" page

**_You can rerun the whole collection and enable the job cache, that would the equivalent action_**
Good point, thanks, I haven't enabled it on my instance yet, I want to test this out soon.

**_this is an entirely different thing that will result in a different output structure that is flattened by one level_**
I understand.

**Comment by vladvisan** [comment]
Also a separate comment:

**"Resume dependencies from this job" even for re-runs of succesful jobs?**
- I had assumed this was the case, but I was testing, and this option only appears for failed jobs (whose associated step has downstream steps)
- At least for some scientists where I work, the option to re-try parts of the workflow from a given step is useful, with slightly different parameters from that step forwards (but with the same datasets before)
- Although this could seemingly also be achieved (assuming job cache is activated) by re-running the whole workflow, and just changing the parameters of that step

**Comment by mvdbeek** [comment]
> * individual datasets

yes, that's right, if you click on rerun there you can replace the single input with the high level input (i.e. the collection input). I agree that this should probably a more direct option in the user interface, but I wanted to point out that you can do this.

**Comment by vladvisan** [comment]
**UI option**
Ah, I see, I was able to select the collection as you indicated, in the re-run screen:
- ![image](https://github.com/galaxyproject/galaxy/assets/11979062/1206067b-5fd7-499b-b062-6d3526bde4ae)
- ![image](https://github.com/galaxyproject/galaxy/assets/11979062/96762a66-6b20-4fde-9927-6fcbc08a1565)
- (t being the name of the collection)

**Basic results**
- All the collection's datasets are regenerated (one job launched per dataset), which is nice
- However, ideally only the failed ones would induce new jobs launched
- I tested this on a collection with a mix of failed/successful dataset jobs, and they were are all regenerated/relaunched

**Advanced results**
When I select the “Resume dependencies from this job?” option, the execution refuses to launch, with the following error screen/message (I crossed out the irrelevant information):
![image](https://github.com/galaxyproject/galaxy/assets/11979062/844d5a92-5f00-4de3-9d2c-5ece818afd8e)

I tested this (on Galaxy version 23.2.2.dev0):
- first test: with all the jobs associated to datasets of the collection, being in the failed state
- other test: with some of the jobs associated to the datasets of the collection, being in the failed state, and others being in the success state

Both cases led to the above error message.
